File: Python/219. Contains Duplicate II.py
class Solution:
    def containsNearbyDuplicate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: bool
        """
        maple = dict()
        for i in range(len(nums)):
            if i > k:
                del maple[nums[i - k - 1]]
            if maple.get(nums[i]) != None:
                return True
            maple[nums[i]] = 1
        return False

File: C++/219. Contains Duplicate II.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        set<int> s;
        int len = nums.size();
        for(int i=0;i<len;i++)
        {
            if(i>k) s.erase(nums[i-k-1]);
            if(s.find(nums[i])!=s.end())
                return true;
            s.insert(nums[i]);
        }
        return false;
    }
};